lib/auth/vault.js

```diff
@@ -33,113 +31,79 @@ const vault = {};
 /**
  * authenticateV2Request
  *
- * @param {string} accessKey - user's accessKey
- * @param {string} signatureFromRequest - signature sent with request
- * @param {string} stringToSign - string to sign built per AWS rules
- * @param {string} algo - either SHA256 or SHA1
-* @param {RequestContext} requestContext - an instance of a RequestContext
+ * @param {string} params - the authentication parameters as returned by
+ *                          auth.extractParams
+ * @param {number} params.version - shall equal 4
+ * @param {number} params.version - shall equal 2
+ * @param {string} params.data.accessKey - the user's accessKey
+ * @param {string} params.data.signatureFromRequest - the signature read from
+ *                                                    the request
+ * @param {string} params.data.stringToSign - the stringToSign
+ * @param {string} params.data.algo - the hashing algorithm used for the
+ *                                    signature
+ * @param {string} params.data.authType - the type of authentication (query or
+ *                                        header)
+ * @param {string} params.data.signatureVersion - the version of the signature
```

Comment:
Given that the signatureVersion looks like a constant shared by all versions of the same generation, this could possibly replace the current "version" attribute. Do you think it would be more relevant ? (this is actually kind of trying to catchup with the mouse that's fleeing but well, we shall reach the end of it someday)